src/main/java/apoc/algo/PageRank.java

```diff
@@ -55,6 +69,34 @@
                 Util.typesAndDirectionsToTypesArray((String) config.getOrDefault(SETTING_PAGE_RANK_TYPES, "")));
     }
 
+    @Procedure("apoc.algo.pageRankWithCypher")
+    @Description(
+            "")
+    public Stream<NodeScore> pageRankWithCypher(
+            @Name("nodes") List<Node> nodes,
+            @Name("config") Map<String, Object> config) {
+        Long iterations = (Long) config.getOrDefault(SETTING_PAGE_RANK_ITERATIONS, DEFAULT_PAGE_RANK_ITERATIONS);
+        String relCypher = (String)config.getOrDefault(SETTING_PAGE_RANK_CYPHER_REL, DEFAULT_PAGE_RANK_CYPHER_REL);
+        String nodeCypher = (String)config.getOrDefault(SETTING_PAGE_RANK_CYPHER_NODE, DEFAULT_PAGE_RANK_CYPHER_NODE);
+
+        boolean shouldWrite = (boolean)config.getOrDefault(SETTING_PAGE_RANK_WRITE, DEFAULT_PAGE_RANK_WRITE);
+
+        if (nodeCypher.equals("") || nodeCypher.isEmpty()) {
+            nodeCypher = DEFAULT_PAGE_RANK_CYPHER_NODE;
+        }
+
+        if (relCypher.equals("") || relCypher.isEmpty()) {
+            relCypher = DEFAULT_PAGE_RANK_CYPHER_REL;
+        }
+
+        PageRankArrayStorageParallelCypher pageRank = new PageRankArrayStorageParallelCypher(dbAPI, pool, nodeCypher, relCypher);
+        pageRank.compute(iterations.intValue());
+        if (shouldWrite) {
+            pageRank.writeResultsToDB();
+        }
+        return nodes.stream().map(node -> new NodeScore(node, pageRank.getResult(node.getId())));
```

Comment:
Yes

Von meinem iPhone gesendet

> Am 11.08.2016 um 15:04 schrieb Atul Jangra <notifications@github.com>:
> 
> In src/main/java/apoc/algo/PageRank.java:
> 
> > +        boolean shouldWrite = (boolean)config.getOrDefault(SETTING_PAGE_RANK_WRITE, DEFAULT_PAGE_RANK_WRITE);
> > +
> > +        if (nodeCypher.equals("") || nodeCypher.isEmpty()) {
> > +            nodeCypher = DEFAULT_PAGE_RANK_CYPHER_NODE;
> > +        }
> > +
> > +        if (relCypher.equals("") || relCypher.isEmpty()) {
> > +            relCypher = DEFAULT_PAGE_RANK_CYPHER_REL;
> > +        }
> > +
> > +        PageRankArrayStorageParallelCypher pageRank = new PageRankArrayStorageParallelCypher(dbAPI, pool, nodeCypher, relCypher);
> > +        pageRank.compute(iterations.intValue());
> > +        if (shouldWrite) {
> > +            pageRank.writeResultsToDB();
> > +        }
> > +        return nodes.stream().map(node -> new NodeScore(node, pageRank.getResult(node.getId())));
> Optionally we could have an option that the node-query also returns degrees.
> Was this for the running of relationship query twice comment?
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
> 
> -- 
> You received this message because you are subscribed to the Google Groups "neo4j-oss-build" group.
> To unsubscribe from this group and stop receiving emails from it, send an email to neo4j-oss-build+unsubscribe@neotechnology.com.
> To post to this group, send email to neo4j-oss-build@neotechnology.com.
> To view this discussion on the web visit https://groups.google.com/a/neotechnology.com/d/msgid/neo4j-oss-build/neo4j-contrib/neo4j-apoc-procedures/pull/83/r74417538%40github.com.